src/Microsoft.Diagnostics.Monitoring.WebApi/IEgressService.cs

```diff
@@ -20,5 +20,9 @@ internal interface IEgressService
             IEndpointInfo source,
             CollectionRuleMetadata collectionRuleMetadata,
             CancellationToken token);
+
+        Task<EgressResult> ValidateProviderAsync(
```

Comment:
Given that `ValidateProvider` and `ValidateProviderAsync` only differ from an interface perspective in that one is async and the other isn't, I think we whouls rename `ValidateProvider` to `ValidateProviderExists` and `ValidateProviderAsync` to `ValidateProviderOptionsAsync` to better convey the difference between their behaviors.